packages/gatsby/src/schema/resolvers.ts

```diff
@@ -221,12 +248,56 @@ export function link<TSource, TArgs>(
     TArgs
   >
 ): GatsbyResolver<TSource, TArgs> {
-  return async function linkResolver(
+  return function linkResolver(
     source,
     args,
     context,
     info
-  ): Promise<any> {
+  ): IGatsbyNode | Array<IGatsbyNode> | null | Promise<any> {
```

Comment:
Sure, I wasn't sure whether resolvers had to always resolve to nodes or not (and whether fixing that wouldn't cause a rippling effect through the typing system). I'll try it. If it sticks it sticks, if it causes too many errors I'll keep it as is because I'm not planning to fix all the things here :)